src/Redux/Rocket/rocketSlice.js

```diff
@@ -0,0 +1,52 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const url = 'https://api.spacexdata.com/v3/rockets';
+
+const getRocket = async () => {
+  const res = await fetch(url);
+  const rockets = await res.json();
+  return rockets;
+};
+
+export const fetchRockets = createAsyncThunk('rockets/fetchRockets', getRocket);
+const rocketSlice = createSlice({
+  name: 'rockets',
+  initialState: {
+    rockets: [],
+    status: 'Rocket_display',
+    error: null,
+  },
+  reducers: {
+    rocketsReserve: (state, action) => {
+      const newState = state.rockets.map((rocket) => {
+        if (rocket.id !== action.payload) return rocket;
+        return { ...rocket, reserved: !rocket.reserved };
+      });
+      const rocketReserve = state;
+      rocketReserve.rockets = newState;
+    },
+  },
+
+  extraReducers(builder) {
+    builder.addCase(fetchRockets.pending, (state) => {
+      const isPending = state;
+      isPending.status = 'loading';
+    });
+    builder.addCase(fetchRockets.fulfilled, (state, action) => {
+      const isFullfilled = state;
+      isFullfilled.status = 'succeeded';
+      const realRocket = [];
+      action.payload.map((rocket) => realRocket.push({
+        id: rocket.id,
+        name: rocket.rocket_name,
+        description: rocket.description,
+        image: rocket.flickr_images[1],
```

Comment:
[Optional]
- [ ] As pointed out by the first reviewer, kindly make sure you use the first image in the array of flickr_images. Which means, you should put index 0 instead of 1 👍